sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -14,21 +14,14 @@ work with sparse matrices and arrays written in Cython.
 from libc.math cimport fabs, sqrt, isnan
 from libc.stdint cimport intptr_t
 
-cimport numpy as cnp
 import numpy as np
 from cython cimport floating
-from ..utils._typedefs cimport float64_t, int32_t, intp_t, uint64_t
+from ..utils._typedefs cimport float64_t, int32_t, int64_t, intp_t, uint64_t
 
-cnp.import_array()
 
-
-# This `integral` fused type is defined to be used over `cython.integral`
-# to only generate implementations for `int{32,64}_t`.
-# TODO: use `{int,float}{32,64}_t` when cython#5230 is resolved:
-# https://github.com/cython/cython/issues/5230
 ctypedef fused integral:
-    int  # int32_t
-    long long  # int64_t
```

Comment:
This is also present here and this might be removed or simplified:

https://github.com/scikit-learn/scikit-learn/blob/2d182e57dd6ded5f66b42bfb9c468f7ac02df237/sklearn/preprocessing/_csr_polynomial_expansion.pyx#L38-L44

Reply:
I don't understand why we would want to use a type that fuses floating point types with integral types in `sklearn/utils/sparsefuncs_fast.pyx`. It seems with either want integer indices or floating point values.

Reply:
My remark was not clear: I meant that the definition of `DATA_t` can be adapted similarly to this new definition of `integral` since https://github.com/cython/cython/issues/5230 has been resolved.

Reply:
This PR is about removing `cnp.import_array` and already quite large. I'd prefer (someone) to deal with csr_polynomial in a separate PR.